### PR TITLE
Add support to new txs

### LINKED ISSUE #90: Support for Staking Allow
Based on https://docs.google.com/document/d/157PmOBAFBCh9NvogkxFr8dLoNE6QB404-0zqHYoa6z8/edit#

### PR DESCRIPTION
**Pull request by emmanuelm41**

### CODE REVIEW

File: app/src/consumer/parser_consumer.c [hunk 1]
```diff
@@ -360,6 +366,69 @@ __Z_INLINE parser_error_t parser_getItemTx(const parser_context_t *ctx,
...
             break;
+        case stakingWithdraw:
+            switch (displayIdx) {
+                case 0: {
+                    snprintf(outKey, outKeyLen, "Type");
+                    *pageCount = 1;
+                    return parser_getType(ctx, outVal, outValLen);
+                }
+                case 1: {
+                    snprintf(outKey, outKeyLen, "Address");
```
Comment (tjanez) [inline app/src/consumer/parser_consumer.c:377]:
```suggestion
                    snprintf(outKey, outKeyLen, "From");
```

File: app/src/consumer/parser_consumer.c [hunk 2]
```diff
@@ -360,6 +366,69 @@ __Z_INLINE parser_error_t parser_getItemTx(const parser_context_t *ctx,
...
+                    return parser_printAddress(&parser_tx_obj.oasis.tx.body.stakingWithdraw.from,
+                                               outVal, outValLen, pageIdx, pageCount);
+                }
+                case 2: {
+                    snprintf(outKey, outKeyLen, "Amount");
+                    return parser_printQuantity(&parser_tx_obj.oasis.tx.body.stakingWithdraw.amount,
+                                                outVal, outValLen, pageIdx, pageCount);
+                }
+                case 3: {
+                    // ??? displayIdx == 1 && parser_tx_obj.oasis.tx.has_fee
```
Comment (tjanez) [inline app/src/consumer/parser_consumer.c:387]: What does this mean?

File: app/src/consumer/parser_consumer.c [hunk 3]
```diff
@@ -360,6 +366,69 @@ __Z_INLINE parser_error_t parser_getItemTx(const parser_context_t *ctx,
...
+                    *pageCount = 1;
+                    return parser_getType(ctx, outVal, outValLen);
+                }
+                case 1: {
+                    snprintf(outKey, outKeyLen, "Beneficiary");
+                    return parser_printAddress(&parser_tx_obj.oasis.tx.body.stakingAllow.beneficiary,
+                                               outVal, outValLen, pageIdx, pageCount);
+                }
+                case 2: {
+                    // FIXME: Handle stakingAllow.is_negative in amount_change
```
Comment (tjanez) [inline app/src/consumer/parser_consumer.c:413]: Yes, please, change the label to "Amount change" and display the sign (i.e. `+` / `-` ) in from of the amount, depending on the value of the `stakingAllow.is_negative` boolean.

Comment (jleni) [reply app/src/consumer/parser_consumer.c:413]: this is not the correct place to do this.. 
Sign handling should go in `parser_printQuantity`

Comment (jleni) [reply app/src/consumer/parser_consumer.c:413]: the reason is.. we need to handle paging correctly and we don't want to sign to be misplaced when doing this

Comment (emmanuelm41) [reply app/src/consumer/parser_consumer.c:413]: Where do we have to handle the sign?

Comment (emmanuelm41) [reply app/src/consumer/parser_consumer.c:413]: I just updated the test cases and now some of them are failing. I have to handle the sign to make them pass.

Comment (emmanuelm41) [reply app/src/consumer/parser_consumer.c:452]: I fixed the tests, and now all is green. I know the way I handle the sign is not the correct. Could you tell me where to handle it?

File: app/src/consumer/parser_consumer.c [hunk 4]
```diff
@@ -360,6 +403,67 @@ __Z_INLINE parser_error_t parser_getItemTx(const parser_context_t *ctx,
...
+                    *pageCount = 1;
+                    return parser_getType(ctx, outVal, outValLen);
+                }
+                case 1: {
+                    snprintf(outKey, outKeyLen, "Beneficiary");
+                    return parser_printAddress(&parser_tx_obj.oasis.tx.body.stakingAllow.beneficiary,
+                                               outVal, outValLen, pageIdx, pageCount);
+                }
+                case 2: {
+                    snprintf(outKey, outKeyLen, "Amount");
```
[the remark this answers is not in the document]
Comment (jleni) [reply app/src/consumer/parser_consumer.c]: This last change probably broke unit tests...
we should not change code without updates to the expected values

[the remark this answers is not in the document]
Comment (tjanez) [reply app/src/consumer/parser_consumer.c]: Sure, I didn't mean to actually just commit this suggestion but to change the label everywhere, also in tests.

[the remark this answers is not in the document]
Comment (emmanuelm41) [reply app/src/consumer/parser_consumer.c]: Done!

**Pull request merged**